backend/setup.go

```diff
@@ -136,30 +156,87 @@ func SetupTracer(pluginID string, tracingOpts tracing.Opts) error {
 		// Initialize global tracer for plugin developer usage
 		tracing.InitDefaultTracer(otel.Tracer(pluginID))
 	}
-	Logger.Debug("Tracing", "enabled", tracingCfg.IsEnabled(), "propagation", tracingCfg.Propagation)
+
+	Logger.Debug(
+		"Tracing",
+		"enabled", tracingCfg.isEnabled(),
+		"propagation", tracingCfg.propagation,
+		"samplerType", tracingCfg.sampler.SamplerType,
+		"samplerParam", tracingCfg.sampler.Param,
+		"samplerRemoteURL", tracingCfg.sampler.Remote.URL,
+		"samplerRemoteServiceName", tracingCfg.sampler.Remote.ServiceName,
+	)
 	return nil
 }
 
 // tracingConfig contains the configuration for OTEL tracing.
 type tracingConfig struct {
-	Address     string
-	Propagation string
+	address     string
+	propagation string
+
+	sampler tracerprovider.SamplerOptions
 }
 
-// IsEnabled returns true if OTEL tracing is enabled.
-func (c tracingConfig) IsEnabled() bool {
-	return c.Address != ""
+// isEnabled returns true if OTEL tracing is enabled.
+func (c tracingConfig) isEnabled() bool {
+	return c.address != ""
 }
 
 // getTracingConfig returns a new tracingConfig based on the current environment variables.
-func getTracingConfig() tracingConfig {
-	var otelAddr, otelPropagation string
+func getTracingConfig(buildInfoGetter build.InfoGetter) tracingConfig {
+	var otelAddr, otelPropagation, samplerRemoteURL, samplerParamString string
+	var samplerType tracerprovider.SamplerType
+	var samplerParam float64
 	otelAddr, ok := os.LookupEnv(PluginTracingOpenTelemetryOTLPAddressEnv)
 	if ok {
+		// Additional OTEL config
 		otelPropagation = os.Getenv(PluginTracingOpenTelemetryOTLPPropagationEnv)
+
+		// Sampling config
+		samplerType = tracerprovider.SamplerType(os.Getenv(PluginTracingSamplerTypeEnv))
+		samplerRemoteURL = os.Getenv(PluginTracingSamplerRemoteURL)
+		samplerParamString = os.Getenv(PluginTracingSamplerParamEnv)
+		var err error
+		samplerParam, err = strconv.ParseFloat(samplerParamString, 64)
+		if err != nil {
+			// Default value if invalid float is provided is 1.0 (AlwaysSample)
+			log.DefaultLogger.Warn(
+				"Could not parse sampler param to float, defaulting to 1.0",
+				"samplerParam", samplerParamString,
```

Comment:
Would be nice to include the actual error as well

Reply:
My bad I'll add that